app/src/androidTest/java/com/sdp/movemeet/Activity/UploadActivityActivityTest.java

```diff
@@ -113,7 +110,7 @@ public void onComplete(@NonNull Task<AuthResult> task) {
     @Test
     @LargeTest
     public void endToEnd() {
-        ActivityScenario scenario = testRule.getScenario();
+        ActivityScenario scenario = ActivityScenario.launch(UploadActivityActivity.class);
```

Comment:
Here, the "testRule.getScenario() " retrieved the scenario created in the @Rule. Except that with my security modifications, instantiating directly a page requiring a connection redirects to the login so no test can be done since we are not on the activity expected by the tests. With this way of doing, there is no more @Rule.We start by connecting in the @Before and only when we are connected, we instantiate the activity we want thanks to "ActivityScenario.launch(UploadActivityActivity.class)"

Reply:
Ok I see, thank your for your explanation!